tests/local/network.go

```diff
@@ -453,74 +434,146 @@ func (n *LocalNetwork) RelayMessage(ctx context.Context,
+		n.tmpnet.Subnets,
+		func(s *tmpnet.Subnet) bool { return s.SubnetID == subnetID },
+	)]
+
+	// consume some of the extraNodes
+	newValidatorNodes := n.extraNodes[0:count]
+	slices.Delete(n.extraNodes, 0, int(count))
+
+	apiURI, err := n.tmpnet.GetURIForNodeID(subnet.ValidatorIDs[0])
 	Expect(err).Should(BeNil())
 
-	// Add the new node names
-	n.subnetNodeNames[subnetID] = append(n.subnetNodeNames[subnetID], nodeNames...)
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	err = subnet.AddValidators(
+		ctx,
+		os.Stdout,
+		apiURI,
+		newValidatorNodes...,
+	)
+	Expect(err).Should(BeNil())
+
+	for _, node := range newValidatorNodes {
+		subnet.ValidatorIDs = append(subnet.ValidatorIDs, node.NodeID)
+		node.Flags[config.TrackSubnetsKey] = subnetID.String()
+	}
+
+	tmpnet.WaitForActiveValidators(ctx, os.Stdout, platformvm.NewClient(n.tmpnet.Nodes[0].URI), subnet)
+
+	nodeIdsToRestart := make([]ids.NodeID, len(newValidatorNodes))
+	for i, node := range newValidatorNodes {
+		nodeIdsToRestart[i] = node.NodeID
+	}
+	n.RestartNodes(ctx, nodeIdsToRestart)
 
 	n.setAllSubnetValues()
 }
 
-// GetAllNodeNames returns a slice that copies all node names in the network
-func (n *LocalNetwork) GetAllNodeNames() []string {
-	// The network starts off with 5 nodes that validate the primary network.
-	// These nodes were not added by this network setup, and are not in n.subnetNodeNames.
-	// So we query the ANR client to get the full list of node names.
-	status, err := n.anrClient.Status(context.Background())
-	Expect(err).Should(BeNil())
-	return status.GetClusterInfo().GetNodeNames()
+// GetAllNodeIDs returns a slice that copies the NodeID's of all nodes in the network
+func (n *LocalNetwork) GetAllNodeIDs() []ids.NodeID {
+	nodeIDs := make([]ids.NodeID, len(n.tmpnet.Nodes))
+	for i, node := range n.tmpnet.Nodes {
+		nodeIDs[i] = node.NodeID
+	}
+	return nodeIDs
 }
 
-func (n *LocalNetwork) RestartNodes(ctx context.Context, nodeNames []string, opts ...runner_sdk.OpOption) {
-	log.Info("Network restarting nodes", "nodeNames", nodeNames)
-	opts = append(opts,
-		runner_sdk.WithExecPath(n.manager.ANRConfig.AvalancheGoExecPath),
-		runner_sdk.WithPluginDir(n.manager.ANRConfig.PluginDir))
-	for _, nodeName := range nodeNames {
-		_, err := n.anrClient.RestartNode(ctx, nodeName, opts...)
+func (n *LocalNetwork) RestartNodes(ctx context.Context, nodeIDs []ids.NodeID) {
+	log.Info("Restarting nodes", "nodeIDs", nodeIDs)
+	var nodes []*tmpnet.Node
+	for _, nodeID := range nodeIDs {
+		for _, node := range n.tmpnet.Nodes {
+			if node.NodeID == nodeID {
+				nodes = append(nodes, node)
+			}
+		}
+	}
+
+	for _, node := range nodes {
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+		err := node.SaveAPIPort()
+		Expect(err).Should(BeNil())
+
+		err = node.Stop(ctx)
+		Expect(err).Should(BeNil())
+
+		err = n.tmpnet.StartNode(ctx, os.Stdout, node)
 		Expect(err).Should(BeNil())
 	}
```

Comment:
How long does this take? If it's long, could we do each iteration of this in a goroutine to speed it up?

Reply:
it's not long. the `StartNode` call here doesn't wait for the node to start up, it's just sort of "fire and forget". 

I found some output in my terminal from the last time I ran this (on my azure dev box), and between the log statements just before and just after this loop, 3.574 had elapsed.

Reply:
also meant to add: note that in the loop just after this one, that's where we wait for all the nodes to come up, and that part in theory takes longer. I don't have a convenient log history to quantify _that_ wait, but I can say that that wait AND a subsequent call to `subnetEvmUtils.IssueTxsToActivateProposerVMFork()` in total took 8.486 seconds.